apps/api/v1/pages/api/event-types/_post.ts

```diff
@@ -275,26 +275,58 @@ async function postHandler(req: NextApiRequest) {
     children: _,
     ...parsedBody
   } = schemaEventTypeCreateBodyParams.parse(body || {});
-
+  if (parsedBody.teamId) {
+    if (parsedBody.parentId) {
+      // parentId is present only for child event type, teamId is used for event types belonging to team
+      throw new HttpError({
+        statusCode: 400,
+        message: "`parentId` and `teamId` both cannot be present in the request",
+      });
+    }
+    if (parsedBody.userId) {
+      //team event is not associated with an owner or user
+      throw new HttpError({
+        statusCode: 400,
+        message: "`teamId` and `userId` both cannot be present in the request",
+      });
+    }
+  } else {
+    if (parsedBody.schedulingType) {
+      //schedulingType is applicable only for team events
+      throw new HttpError({
+        statusCode: 400,
+        message: "schedulingType is applicable only for team events",
+      });
+    }
+  }
   let data: Prisma.EventTypeCreateArgs["data"] = {
     ...parsedBody,
-    userId: !!parsedBody.teamId ? null : userId,
-    users: !!parsedBody.teamId ? undefined : { connect: { id: userId } },
     bookingLimits: bookingLimits === null ? Prisma.DbNull : bookingLimits,
     durationLimits: durationLimits === null ? Prisma.DbNull : durationLimits,
   };
 
+  if (!!parsedBody.teamId) {
+    data = {
+      ...data,
+      userId: null,
+      users: undefined,
+    };
+  } else {
+    data = {
+      ...data,
+      // Connect current user only if it is not a managed eventType, or the current user is not systemWideAdmin
+      userId: parsedBody.parentId || isSystemWideAdmin ? parsedBody.userId : userId,
+      users: { connect: { id: parsedBody.parentId || isSystemWideAdmin ? parsedBody.userId : userId } },
```

Comment:
You can create eventType for other users only if the parentId is present, i.e you are assigning hosts for managed event type, or current user is systemWideAdmin. Since we are performing checks for both cases later in the code, I haven't added any checks here.